pkg/registry/generic/registry/storage_factory.go

```diff
@@ -30,7 +30,12 @@ func StorageWithCacher(
 	objectType runtime.Object,
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
-	newListFunc func() runtime.Object) storage.Interface {
+	newListFunc func() runtime.Object,
+	// TODO: Currently we are passing 0 or 1 indices now (more than
```

Comment:
I think we shouldn't merge with this TODO, I really don't like using variadic in this way.

But also I was thinking, what if you added something to the matcher itself that could publish this? I don't think you really need this information at construction time.